lib/libversion.py

```diff
@@ -275,6 +292,13 @@ def __init__(self, version_specs):
         self.specs = [(VersionSpecifier.OPS[op], version) for (op, version) in version_specs]
         self.str = str(version_specs)
 
+        self.specifier = None
+        if HAS_SPECIFIER:
+            try:
+                self.specifier = SpecifierSet(''.join(version_specs[0]))
+            except:
```

Comment:
Do not use bare `except:`, it also catches unexpected events like memory errors, interrupts, system exit, and so on.
* [PEP8}(https://peps.python.org/pep-0008/#programming-recommendations)
* https://realpython.com/the-most-diabolical-python-antipattern